en/computer_vision/obstacle_avoidance.md

```diff
@@ -0,0 +1,156 @@
+
+
+## Mission Mode Avoidance {#mission_mode}
+
+PX4 supports obstacle avoidance in [Missions](../flight_modes/mission.md), using avoidance software running on a separate companion computer. 
+
+Obstacle avoidance is enabled within PX4 by [setting](../advanced_config/parameters.md) the `MPC_OBS_AVOID` to 1.
+PX4 communicates with the obstacle avoidance software using an implementation of the MAVLink [Path Planning Protocol](TBD) (Trajectory Interface) which is [#described below](mission_avoidance_interface).
+Provided an avoidance system complies with this interface it can be used with PX4.
+
+The tested companion computer platform is [Intel Aero](../flight_controller/intel_aero.md) running either the *local_planner* or *global_planner* avoidance software.
+This is set up as described in the [Intel Aero > Obstacle Avoidance](../flight_controller/intel_aero.md#obstacle-avoidance) and in the [PX4/avoidance](https://github.com/PX4/avoidance#obstacle-detection-and-avoidance) Github repo.
+
+
+### Mission Progression
+
+The companion computer provides setpoint navigation for the *entire route* (not just when there are obstacles) based on current position, current waypoint and next waypoint information from PX4.
+
+The resulting mission behaviour is very similar to missions without obstacle avoidance, and mainly affects the criteria used to determine that a waypoint has been reached.
+
+The difference when avoidance is active are:
+- A waypoint is "reached" when the vehicle is within the acceptance radius, regardless of its heading.
+  - In normal missions the vehicle must reach a waypoint with a certain heading (i.e. in a "close to" straight line from the previous waypoint). 
+  - When obstacle avoidance is active, this constraint cannot be fulfilled because the obstacle avoidance algorithm has full control of the vehicle heading, and the vehicle always moves in the current field of view. 
+- navigator updates the triplets when the vehicle has reached the acceptance radius of each waypoint. 
+  If a waypoint is inside an obstacle it can happen that it’s never reach and the mission will be stuck. 
+  If the vehicle projection on the line previous-current waypoint has passed the current waypoint, the acceptance radius is enlarged such that the current waypoint is set as reached
+- If the vehicle within the x-y acceptance radius, the altitude acceptance is modified such that the mission progresses (even if it is not in the altitude acceptance radius).
+  
+
+###  Mission Mode Avoidance Interface {#mission_avoidance_interface}
+
+Mission mode is enabled on PX4 by setting `MPC_OBS_AVOID` to `1`.
+
+PX4 communicates with the computer vision system on the companion computer using an implementation of the [Path Planning Protocol](TBD) (Trajectory Interface). <!-- check we call it this! -->
+
+PX4 sends the desired trajectory to the companion computer in `TRAJECTORY_REPRESENTATION_WAYPOINTS` messages at 5Hz.
+The "waypoint" array fields are set as shown:
```

Comment:
@mrivi is the uOrb topic for the desired trajectory a 1:1 mapping to the TRAJECTORY_REPRESENTATION_WAYPOINTS? (this is just your text for the uOrb copied as though it was also true for the TRAJECTORY_REPRESENTATION_WAYPOINTS)

I wanted to talk about the interface separate from the implementation.

Reply:
No, they aren't. Here I have described [this array](https://github.com/PX4/Firmware/blob/e91db7b4d2694cd7a1c11b07e979fbbbb5561e99/msg/vehicle_trajectory_waypoint.msg#L19)